pkg/agent/flowexporter/priorityqueue.go

```diff
@@ -20,12 +20,15 @@ import (
 )
 
 type ItemToExpire struct {
-	// Flow related info
-	ConnKey          *ConnectionKey
-	FlowRecord       *FlowRecord
-	Conn             *Connection
-	ActiveExpireTime time.Time
-	IdleExpireTime   time.Time
+	ConnKey                              ConnectionKey
+	Conn                                 *Connection
```

Comment:
I actually went through several back and forth. I think there are three reasons make me tend to keep a pointer instead of value copy.
1. We don't want to export idle time out records multiple times, so we have to temporarily remove those records from the PQ after the 1st export(also set`conn.PriorityQueueItem = nil` in [line](https://github.com/antrea-io/antrea/blob/4b2f88c6e8e40403cde67d13219fa823063724a4/pkg/agent/flowexporter/exporter/exporter.go#L289) ). And then add them back in `AddOrUpdateConn` when we verify it is active and reset the idle time out. The thing is, we cannot tell if the pqItem is still in the PQ or not if we stores a copy of the conn. We need to check whether `conn.PriorityQueueItem` is `nil` in [line](https://github.com/antrea-io/antrea/blob/4b2f88c6e8e40403cde67d13219fa823063724a4/pkg/agent/flowexporter/connections/conntrack_connections.go#L233). If it is in the PQ, we call `pq.Update`, if not, we create a new pqItem and add it into the PQ.
2. I think if we make both `AddOrUpdateConn` in conntrack_connections.go and `SendFlowRecords` in exporter.go, access connection map lock first, and access priority queue lock later, this will avoid deadlock. I mean given two process A&B, need to access both resources c&d, if we keep the order of resources acquire to be the same in A&B, then we will not introduce deadlock?
3. Now we are also modifying deny connection map in `sendFlowRecord`, if we want to avoid doing that, we probably need to add more flag into pqItem struct. I feel that's a bit cumbersome for the struct.

Reply:
please just ignore my last comment. I think I'll just go back to use value copy. As we cannot set `conn.PriorityQueueItem` to `nil`, we need to use [line](https://github.com/heanlan/antrea/blob/4b2f88c6e8e40403cde67d13219fa823063724a4/pkg/agent/flowexporter/connections/conntrack_connections.go#L229) to check stats change.